### ISSUE TITLE
Issue upgrading Docker Desktop for windows from v4.16.1 to v4.16.2

### ISSUE THREAD

**Issue by jeffrozica** [docker/for-win#13183]
I just tried to upgrade docker desktop for windows from v4.16.1 to v4.16.2 on windows 11 and its stuck in updating and restart, normally it would run for a few minutes there, then kick off another screen where  you can see it upgrading, but it's hung.  I think I'm going to kill it, but that means reinstalling again and loosing all of my docker images again.

I had a terrible time getting to v4.16.1 and now here we go again, This v4.16.x is the worst docker upgrade ever. Very frustrated.

**Comment by YouveGotMeowxy** [comment]
I had a similar issue. It just seemed to hang while the tooltip and animation for the tray icon was showing "stopping docker desktop...".

I eventually just rebooted anyway, and when windows loaded DD loaded fine, and is up to date, all containers running, but I can no longer access any of them. Like something in the network changed or something. I'm not really a network person so not real good at diagnosing this kinda thing.

**Comment by diablodale** [comment]
Similar failures with v4.16.2 upgrade.
v4.16.1 would never "shutdown" for the upgrade.
Reboot Windows
Then DD could upgrade itself from `4.16.1` to `()`.
4.16.2 installed
and then it never restarts. It continuously stuck in "docker desktop starting".
Windows service mgr shows Docker service is running.
WSL2 is working
but no dcker in wsl2 and no response from DD ui.
Killed DD. Stopped windows service.
Windows-I, uninstalled DD
Downloaded today's v4.16.2 installer for Windows x64
installed
Ran DD from windows menu
Stuck forever in "docker desktop starting".
Seems the regressions continue :-(

**Comment by p1-0tr** [comment]
hi @diablodale , the issue running 4.16.2 encountering is likely related to https://github.com/docker/for-win/issues/13184 . I posted a dev build on that ticket, if you are willing to try it please let me know how it goes. Should the problem persist, could you collect diagnostics - https://docs.docker.com/desktop/troubleshoot/overview/#diagnose-from-the-terminal (I am most interested in %LOCALAPPDATA%\Docker\log.txt).

**Comment by diablodale** [comment]
Yes, agreed this issue looks like a dup of the issue you link. I'll convo over on that issue
